### ISSUE TITLE
pytorch syntax question.

### ISSUE THREAD

**Issue by herrickli** [amdegroot/ssd.pytorch#403]
In the 82 line of `mulitibox_loss.py`, the code is as follows:
```
        pos_idx = pos.unsqueeze(pos.dim()).expand_as(loc_data)
        loc_p = loc_data[pos_idx].view(-1, 4)
        loc_t = loc_t[pos_idx].view(-1, 4)
```
The shape of `loc_data` and `pos_idx` is the same, but use syntax `loc_data[pos_idx]` doesn't expand the shape of `loc_p`, instead it become a one dim Tensor, I am confused about this syntax, beause in python terminal, if I set 
```
b = torch.randn(2,2,4)
mask = torch.zeros(2,2,4)．long()
b[mask].shape #torch.Size([2, 2, 4, 2, 4])
``` 
I simulate the `loc_data[pos_idx]` with `b[mask]`, apparently the results are not in the same form.
So, could someone help me with understanding this syntax please, very thanks.
